shoop/core/models/_shipments.py

```diff
@@ -45,12 +48,19 @@ class Shipment(models.Model):
     description = models.CharField(max_length=255, blank=True, verbose_name=_("description"))
     volume = MeasurementField(unit="m3", verbose_name=_("volume"))
     weight = MeasurementField(unit="kg", verbose_name=_("weight"))
+    identifier = InternalIdentifierField(unique=True)
     # TODO: documents = models.ManyToManyField(FilerFile)
 
     class Meta:
         verbose_name = _('shipment')
         verbose_name_plural = _('shipments')
 
+    def __init__(self, *args, **kwargs):
+        super(Shipment, self).__init__(*args, **kwargs)
+        if not self.identifier:
+            self.identifier = "%s/%s/%s" % (
+                self.order.pk, self.order.shipments.count(), random.randint(1, 1000000000))
```

Comment:
While at it, replace the random.randint with the thing from django that generates random strings securely...